bytes_ostream.hh

```diff
@@ -154,19 +144,21 @@ private:
             if (!space) {
                 throw std::bad_alloc();
             }
-            auto new_chunk = std::unique_ptr<chunk>(new (space) chunk());
-            new_chunk->offset = size;
-            new_chunk->size = alloc_size - sizeof(chunk);
-            if (_current) {
-                _current->next = std::move(new_chunk);
-                _current = _current->next.get();
-            } else {
-                _begin = std::move(new_chunk);
-                _current = _begin.get();
-            }
+            auto backref = _current ? &_current->next : &_begin;
+            auto new_chunk = new (space) chunk(backref, alloc_size - sizeof(chunk), size);
+            _current = new_chunk;
             _size += size;
             return _current->data;
     }
+    [[gnu::noinline]]
+    void free_chain(chunk* c) {
```

Comment:
noexcept

Reply:
yes, will update.

Note that noexcept isn't too helpful on inline functions. But:
1. this is called from a destructor, so it serves some documentary purpose
2. this is a good candidate for deinlining later